[[
Prevent blinking cursor
Author: Peer
v1:
- prevent blinking cursor early on GRUB init
v2:
- leave blinking cursor hidden after exiting menu
]]

diff -urp a/grub-core/kern/main.c b/grub-core/kern/main.c
--- a/grub-core/kern/main.c	2017-03-10 14:12:25.874247996 +0100
+++ b/grub-core/kern/main.c	2017-03-10 14:35:21.661396054 +0100
@@ -267,6 +267,8 @@ grub_main (void)
 {
   /* First of all, initialize the machine.  */
   grub_machine_init ();
+  /* Disable the cursor, early in the boot process */ 
+  grub_setcursor (0); 
 
   grub_boot_time ("After machine init.");
 
diff -urp a/grub-core/kern/term.c b/grub-core/kern/term.c
--- a/grub-core/kern/term.c	2017-03-10 14:12:25.874247996 +0100
+++ b/grub-core/kern/term.c	2017-03-10 14:35:21.661396054 +0100
@@ -159,6 +159,15 @@ grub_getkey (void)
   return 0;
 }
 
+void 
+grub_setcursor (int on) 
+{ 
+  struct grub_term_output *term; 
+
+  FOR_ACTIVE_TERM_OUTPUTS(term) 
+    grub_term_setcursor (term, on); 
+} 
+
 void
 grub_refresh (void)
 {
diff -urp a/grub-core/normal/menu_text.c b/grub-core/normal/menu_text.c
--- a/grub-core/normal/menu_text.c	2017-03-01 13:53:39.635958037 +0100
+++ b/grub-core/normal/menu_text.c	2017-03-10 14:37:16.326332135 +0100
@@ -517,7 +517,6 @@ menu_text_fini (void *dataptr)
 {
   struct menu_viewer_data *data = dataptr;
 
-  grub_term_setcursor (data->term, 1);
   grub_term_cls (data->term);
   grub_free (data);
 }
diff -urp a/include/grub/term.h b/include/grub/term.h
--- a/include/grub/term.h	2017-03-10 14:12:25.874247996 +0100
+++ b/include/grub/term.h	2017-03-10 14:35:21.661396054 +0100
@@ -332,6 +332,7 @@ void grub_putcode (grub_uint32_t code, s
 int EXPORT_FUNC(grub_getkeystatus) (void);
 int EXPORT_FUNC(grub_key_is_interrupt) (int key);
 void grub_cls (void);
+void EXPORT_FUNC(grub_setcursor) (int on); 
 void EXPORT_FUNC(grub_refresh) (void);
 void grub_puts_terminal (const char *str, struct grub_term_output *term);
 struct grub_term_coordinate *grub_term_save_pos (void);
